qcodes/dataset/data_set.py

```diff
@@ -764,7 +764,7 @@ def _ensure_dataset_written(self) -> None:
 
     def get_parameter_data(
         self,
-        *params: Union[str, ParamSpec, "_BaseParameter"],
+        *params: Union[str, ParamSpec, "ParameterBase"],
```

Comment:
@jenshnielsen should we use python's futures annotations import in the files that use this style of type checking so we can remove the strings?

Reply:
Perhaps, it would be nice since we could

1 remove most string type types
2 replace imports from tying with the regular collections List to list
3 use the shorthand syntax a | b rather than Union[a, b]

without waiting for this to be the default in 3.10

the reason we have been holding back is that it’s still not clear if this really becomes the default https://lwn.net/Articles/858576/